sr_utilities_common/src/sr_utilities_common/aws_manager.py

```diff
@@ -39,25 +37,24 @@ def __init__(self):
         try:
             with open('/usr/local/bin/customer.key', 'r') as customer_key_file:
                 customer_key = customer_key_file.read()
-                headers = {'x-api-key': '{}'.format(customer_key[:-1])}
-        except Exception:
+                headers = {'x-api-key': f'{customer_key[:-1]}'}
+        except IOError:
             rospy.logerr("Could not find customer key, ask software team for help!")
 
         try:
             response = requests.get('https://5vv2z6j3a7.execute-api.eu-west-2.amazonaws.com/prod', headers=headers)
-
             if response.status_code != 200:  # Code for success
+                rospy.logerr(f"Could not connect to AWS API server. Returned status code {response.status_code}")
                 raise Exception()
+        except requests.exceptions.RequestException as e:
+            rospy.logerr(f"Could not request secret AWS access key, ask software team for help!\nError message: {e}")
 
-            result = re.search('ACCESS_KEY_ID=(.*)\nSECRET_ACCESS', response.text)
-            aws_access_key_id = result.group(1)
-            result = re.search('SECRET_ACCESS_KEY=(.*)\nSESSION_TOKEN', response.text)
-            aws_secret_access_key = result.group(1)
-            result = re.search('SESSION_TOKEN=(.*)\nEXPIRATION', response.text)
-            aws_session_token = result.group(1)
-
-        except Exception:
-            rospy.logerr("Could not request secret AWS access key, ask software team for help!")
+        result = re.search('ACCESS_KEY_ID=(.*)\nSECRET_ACCESS', response.text)
+        aws_access_key_id = result.group(1)
+        result = re.search('SECRET_ACCESS_KEY=(.*)\nSESSION_TOKEN', response.text)
+        aws_secret_access_key = result.group(1)
+        result = re.search('SESSION_TOKEN=(.*)\nEXPIRATION', response.text)
+        aws_session_token = result.group(1)
```

Comment:
if requests.get('https://5vv2z6j3a7.execute-api.eu-west-2.amazonaws.com/prod', headers=headers) fails, what's the type of response? 

Probably would be better to move the reassignment of aws_ to the try/except clause